include/crpropa/magneticField/UF23Field.h

```diff
@@ -0,0 +1,153 @@
+#ifndef _UF23Field_h_
+#define _UF23Field_h_
+
+#include <vector>
+#include "crpropa/magneticField/MagneticField.h"
+
+namespace crpropa {
+/**
+ * \addtogroup MagneticFields
+ * @{
+ */
+
+
+/**
+ @class UF23Field
+ @brief UF23Field Galactic magnetic field model
+
+ Implements the eight coherent magnetic field models of UF23
+ See: M. Unger and G.R. Farrar, arXiv:2311.12120
+
+ Assumes a galactocentric coordinate system with the Galactic center
+ at the origin, the x-axis pointing in the opposite direction of the
+ Sun, and the z-axis pointing towards Galactic North.
+
+ */
+
+class UF23Field : public MagneticField  {
+public:
+  /// model variations (see Tab.2 of UF23 paper)
+  enum ModelType {
+    base,
+    neCL,
+    expX,
+    spur,
+    cre10,
+    synCG,
+    twistX,
+    nebCor
+  };
+
+
+public:
+  /**
+     @brief constructor
+     @param mt model type  (see Tab.2 of UF23 paper)
+     @param maxRadiusInKpc maximum radius of field in kpc
+  */
+  UF23Field(const ModelType mt);
+  /// no default constructor
+  UF23Field() = delete;
```

Comment:
You probably chose to not provide a default model because none of the models yielded a significantly better fit than the others, right?

Reply:
Yes, I think that not providing a default model makes the user aware that this is a collection of 8 models to choose from.